src/printer/printer.h

```diff
@@ -87,6 +87,8 @@ class Printer
   virtual void toStreamCmdDeclareFunction(std::ostream& out,
                                           const std::string& id,
                                           TypeNode type) const;
+  /** Variant of above for a pre-existing variable */
+  void toStreamCmdDeclareFunction(std::ostream& out, const Node& v) const;
```

Comment:
I wonder if there is any need for the original methods if these variants are more convenient.

Reply:
It's possible, I'll check.

Reply:
There is one instance where this command is called now (via Command::toStream).

However, the reason why we want the original is because it is easier to implement: the derived printer does not need to know how to extract the name from the variable.

Regardless, I've updated Command to use the new version.